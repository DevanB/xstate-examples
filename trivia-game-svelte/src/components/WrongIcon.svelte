<div class="wrongIcon">
  <svg viewBox='0 0 20 20'>
    <path
      d='M10 8.586L2.929 1.515 1.515 2.929 8.586 10l-7.071 7.071 1.414 1.414L10 11.414l7.071 7.071 1.414-1.414L11.414 10l7.071-7.071-1.414-1.414L10 8.586z'
      fill='currentColor'
    />
  </svg>
</div>

<style>
.wrongIcon {
  @apply h-4 w-4 text-red-500;
}

.wrongIcon:hover {
  @apply text-red-700;
}
</style>
