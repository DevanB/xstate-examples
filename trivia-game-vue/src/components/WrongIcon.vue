<template>
  <p>wrong</p>
</template>

<script>
export default {
  name: "WrongIcon"
}
</script>
