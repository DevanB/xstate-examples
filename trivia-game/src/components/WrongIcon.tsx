import React from 'react'
import tw from 'tailwind.macro'

const CloseSvg = () => (
  <svg viewBox='0 0 20 20'>
    <path
      d='M10 8.586L2.929 1.515 1.515 2.929 8.586 10l-7.071 7.071 1.414 1.414L10 11.414l7.071 7.071 1.414-1.414L11.414 10l7.071-7.071-1.414-1.414L10 8.586z'
      fill='currentColor'
    />
  </svg>
)

const Icon = tw.div`h-4 w-4 text-red-500 hover:text-red-700`

export const WrongIcon = () => (
  <Icon>
    <CloseSvg />
  </Icon>
)
