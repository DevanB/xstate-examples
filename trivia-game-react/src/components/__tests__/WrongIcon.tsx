import {render} from '@testing-library/react'
import React from 'react'
import {WrongIcon} from '../WrongIcon'

describe('<WrongIcon/>', () => {
  it('should match snapshot', () => {
    const {container} = render(<WrongIcon />)
    expect(container).toMatchSnapshot()
  })
})
