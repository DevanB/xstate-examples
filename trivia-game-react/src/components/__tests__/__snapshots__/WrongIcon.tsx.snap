// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WrongIcon/> should match snapshot 1`] = `
<div>
  <div
    class="WrongIcon__Icon-sc-173ovh4-0 cDCmhM"
  >
    <svg
      viewBox="0 0 20 20"
    >
      <path
        d="M10 8.586L2.929 1.515 1.515 2.929 8.586 10l-7.071 7.071 1.414 1.414L10 11.414l7.071 7.071 1.414-1.414L11.414 10l7.071-7.071-1.414-1.414L10 8.586z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;
